src/ParseServer.js

```diff
@@ -143,6 +143,11 @@ class ParseServer {
     enableSingleSchemaCache = false,
     __indexBuildCompletionCallbackForTests = () => {},
   }) {
+    // verify parse-server is running on node >= 4.6
+    if (process.versions.node < '4.6') {
```

Comment:
@flovilmart I whipped up a little something for comparing the semantic versions, assuming both are strictly numeric (but I could expand on that). Assuming this looks sufficient I can work on another PR with this properly added in somewhere. This time, with it being a testable unit, I can add cases to ensure appropriate behavior.
```js
function isSemanticVersionLessThanVersion(version1, version2) {
	var versionParts1 = version1.split(".");
	var versionParts2 = version2.split(".");
	
	for(var x = 0; x < versionParts2.length; x++) {
		if(x > versionParts1.length - 1) {
			// version 1 is shorter than version 2, and thus less than
			return true;
			
		}
		
		// get same parts from both versions to compare
		var part1 = parseInt(versionParts1[x]);
		var part2 = parseInt(versionParts2[x]);
	
		if(part1 < part2) {
		    // version 1 less than version 2
			return true;
			
		} else if(part1 > part2) {
			// version 1 greater than version 2
			return false;
			
		}
	}
	
	// greater than or equal to
	return false;
	
}
```